monai/transforms/spatial/array.py

```diff
@@ -286,7 +294,7 @@ def __call__(  # type: ignore
         return img
```

Comment:
Hi @wyli , I'm not sure but I think lines from 288 to 293 may not correct when `lazy_evaluation = True`.
I tried an example like:
```
tr_lazy = ResampleToMatch()
tr_lazy.lazy_evaluation = True
pending_im_mod = tr_lazy(data["im2"], data["im1"])
``` 
and `pending_im_mod.peek_pending_affine()` does not produce expected result.